### PR TITLE
Getting rid of LayerActions

### PR DESCRIPTION
**Pull request by ayrton04**
FYI @skaynama @p6chen. We are apparently removing LayerActions, so for compatibility with our other changes, I removed any methods that make use of them here. This branch is needed to work with the other internal repos I've been changing. Note that I am merging into a "catch-all" branch instead of indigo-devel for now.

@jasonimercer, feel free to ignore. Just tagging you so you're aware when you return.

### CODE REVIEW

File: costmap_2d/plugins/inflation_layer.cpp [hunk 1]
```diff
@@ -69,7 +67,6 @@ InflationLayer::InflationLayer()
   , cached_costs_(NULL)
   , cached_distances_(NULL)
 {
-  last_known_enabled_ = false;
   access_ = new boost::shared_mutex();
   
   algorithmSelect_.setMaxAlgorithmTypes(ALG_TOTAL_AVAILABLE);
```
Comment (skaynama) [inline costmap_2d/plugins/inflation_layer.cpp]: Should this go away given that we now do on-demand inflation? Or do we still want to do algorithm select for the local costmap inflation?

Comment (ayrton04) [inline costmap_2d/plugins/inflation_layer.cpp]: There are going to have to be multiple PRs for removing these things. This isn't going to be our inflation layer anymore regardless, so it's just a matter of getting it back to its upstream-friendly state and moving any necessary functionality over. I'll let @jasonimercer weigh in as well.

Comment (jasonimercer) [inline costmap_2d/plugins/inflation_layer.cpp]: I think this should revert to as close to upstream as possible. Eventually we will no longer use this code so no need for custom deltas. 

Presently we only use this for local inflation so algorithm picking isn't as important.

Comment (ayrton04) [inline costmap_2d/plugins/inflation_layer.cpp]: Agreed. What I'm trying to do is break this into two stages: in the first stage, I remove/revert only those things in navigation that will cause our other changes to not build or function. In the second stage, I will revert everything else, and pull over our changes where applicable. What do we think?

Comment (skaynama) [inline costmap_2d/plugins/inflation_layer.cpp]: Makes sense. LGTM.

Comment (ayrton04) [inline costmap_2d/plugins/inflation_layer.cpp]: @p6chen, does this look OK to you? I want to get everything in a state where people can start testing on it. This PR isn't merging into indigo-devel, so it won't affect anything if it there are any residual bugs.

**Comment by p6chen** [comment]
LGTM.

**Pull request merged**